js_modules/dagit/packages/ui/src/components/ConfigTypeSchema.tsx

```diff
@@ -186,13 +186,131 @@ export const ConfigTypeSchema = React.memo((props: ConfigTypeSchemaProps) => {
+    // stack before the leaving element as hovered
+
+    let currentHoveredStack: Array<{setHovered: (hovered: boolean) => void}> = [];
+    function useDictEntryHover() {
+      const [hovered, setHovered] = React.useState(false);
+      const self = React.useMemo(() => ({setHovered}), []);
+      return {
+        hovered,
+
+        // Unset the previous hovered target and set the current one
+        onMouseEnter: React.useCallback(() => {
+          const lastHovered = currentHoveredStack[currentHoveredStack.length - 1];
+          if (lastHovered) {
+            // If there is already a hovered element, unhover it.
+            lastHovered.setHovered(false);
+          }
+          // Record that we're now the last entry to be hovered
+          currentHoveredStack.push(self);
+          setHovered(true);
+        }, [self]),
+
+        // Unset the current hovered target and use its parent as the next hovered target if it has one
+        onMouseLeave: React.useCallback(() => {
+          const lastHovered = currentHoveredStack[currentHoveredStack.length - 1];
+          if (!lastHovered) {
+            // This should never happen since we can't MouseLeave something we never MouseEnter'd
+            return;
+          }
+          // We should be the last hovered element, if not lets proceed anyways because
+          // we'll hover the correct element later.
+          lastHovered.setHovered(false);
+
+          // Find the index of this element and remove it.
+          // There shouldn't be anything after it since MouseLeave events should bubble upwards
+          const currentIndex = currentHoveredStack.indexOf(self);
+          if (currentIndex !== -1) {
+            // This should only remove 1 entry, the last hovered entry
+            currentHoveredStack = currentHoveredStack.slice(0, currentIndex);
+          }
+
+          // If something is still on the stack after this dict entry is no longer hovered then
+          // its a parent dict entry and should be hovered
+          const nextLastHovered = currentHoveredStack[currentHoveredStack.length - 1];
+          if (nextLastHovered) {
+            nextLastHovered.setHovered(true);
+          }
+        }, [self]),
+      };
+    }
+    return {useDictEntryHover};
+  }, []);
+  return (
+    <HoveredDictEntryContext.Provider value={value}>{children}</HoveredDictEntryContext.Provider>
+  );
+});
+
+const DictEntry = React.forwardRef(
+  (
+    props: React.ComponentProps<typeof DictEntryDiv>,
+    ref: React.ForwardedRef<HTMLButtonElement>,
+  ) => {
+    const {hovered, onMouseEnter, onMouseLeave} = React.useContext(
+      HoveredDictEntryContext,
+    ).useDictEntryHover();
+
+    return (
+      <DictEntryDiv2>
+        <DictEntryDiv
+          {...props}
+          $hovered={hovered}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+          ref={ref}
+        />
+      </DictEntryDiv2>
+    );
+  },
+);
+
+const DictEntryDiv2 = styled.div``;
```

Comment:
I needed this second div because the `$hovered` branching in the css was causing the class name on the div to keep changing which meant using `&` would not reference the non-hovered versions since they have a different css class name.